### ISSUE TITLE
How can I modify RNN cell weight during each training epoch?

### ISSUE THREAD

**Issue by LiangKlausQiu** [tensorflow/tensorflow#8382]
I want to binarize the weights of RNN-GRU cell during each training epoch, in order to reduce the model size and increase the performance. Instead of binarizing the weights after freezing the graph, I wonder how can I get access to and modify the weights during each training epoch, or more specifically, before computing gradients and updating weights? It seems no API is provided to let RNN cell weight exposed to users.
Thank you so much.

**Comment by MicaelCarvalho** [comment]
Your question can be better addressed on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow). The GitHub "Issues" section is mainly for bug reporting and feature requests.

On the topic: take a look at `tf.get_variable`, it may help you.

**Comment by drpngx** [comment]
As @MicaelCarvalho this is best answered on StackOverflow where we monitor all questions with the tag `tensorflow`. All optimizers have the `compute_gradients` method. You will get the variables and gradients back, then you can modify them at your leisure, before calling `apply_gradients`.
